Mobile/ShopLens/Droid/Activities/CameraActivity.cs

```diff
@@ -187,127 +176,113 @@ protected override void OnActivityResult(int requestCode, Result resultCode, Int
         private void PictureShowBitmap(Uri uri)
         {
             // Put image in gallery.
-            Intent mediaScanIntent = new Intent(Intent.ActionMediaScannerScanFile);
+            var mediaScanIntent = new Intent(Intent.ActionMediaScannerScanFile);
             mediaScanIntent.SetData(uri);
             SendBroadcast(mediaScanIntent);
 
             // Conversion. 
-            int height = ImgView.Height;
-            int width = ImgView.Width;
+            var height = imgView.Height;
+            var width = imgView.Width;
             var image = MediaStore.Images.Media.GetBitmap(ContentResolver, uri);
             image = BitmapHelper.ScaleDown(image, Math.Min(height, width));
-            ImgView.SetImageBitmap(image);
+            imgView.SetImageBitmap(image);
         }
 
         private void RecogniseImage(Uri uri)
         {
-            // Run the image recognition task
-            int maxWebClassifierImageSize = int.Parse(ConfigurationManager.AppSettings["webClassifierImgSize"]);
-            
             progressBar.Visibility = ViewStates.Visible;
             Task.Run(async () =>
-            {
-                var image = MediaStore.Images.Media.GetBitmap(ContentResolver, uri);
-                image = BitmapHelper.ScaleDown(image, maxWebClassifierImageSize);
-                using (var stream = new MemoryStream())
                 {
-                    // 0 because compression quality is not applicable to .png
-                    image.Compress(Bitmap.CompressFormat.Png, 0, stream);
+                    var image = GetResizedImageForClassification(uri);
 
-                    var classificator = DependencyInjection.Container.Resolve<IAsyncImageClassificator>();
-                    var results = await classificator.ClassifyImageAsync(stream.ToArray());
+                    var classifyImageTask = GetClassifyImageTask(image);
 
-                    prefs = new ActivityPreferences(this, PREFS_NAME);
-                    guess = results.OrderByDescending(x => x.Value).First().Key;
-                    
-                    tts.Speak(
-                        $"This is. {guess}",
-                        QueueMode.Flush,
-                        null,
-                        null);
+                    if (!textRecognizer.IsOperational)
+                    {
+                        System.Diagnostics.Debug.WriteLine("[Warning]: Text recognizer not operational.");
+                        return new RecognitionResult { Predictions = await classifyImageTask };
+                    }
                     
-                }
-            })
+                    var recognizeTextTask = GetTextFromImageAsync(image);
+
+                    await Task.WhenAll(classifyImageTask, recognizeTextTask);
```

Comment:
Why is this line needed?

Reply:
We have 2 running tasks at this point which are not awaited